tests/controllers/test_uaa_controller.py

```diff
@@ -282,3 +285,34 @@ def test_reset_user_password_by_id_password_change_success(self, mock_request):
             output = uaa_controller.change_user_password_by_id(user_id, user_password)
             self.assertEqual(output.status_code, 200)
             self.assertEqual(output.json(), password_change_success_json)
+
+    @requests_mock.mock()
+    def test_successful_processing_of_permissions_expiry_with_a_iso8601_date(self, mock_request):
```

Comment:
We realistically are testing the _If_ is processed correctly, the result of which is refresh is called. With that in mind we should mock that function and confirm just that. At the moment the test is generic I haven't broken. It also means we can get rid of the other mocks, something like 

```    
   @patch("response_operations_ui.controllers.uaa_controller.refresh_permissions")
    def test_user_has_permission_refresh_permissions_isoformat(self, refresh_permissions):
        
        with self.app.test_request_context():
            session["permissions"] = {
                "groups": test_groups,
                "expiry": datetime.isoformat(datetime.now() + timedelta(minutes=-5)),
            }
            uaa_controller.user_has_permission("oauth.approvals", user_id)
            refresh_permissions.assert_called_with(user_id)

Reply:
The function name(s) should also reflect the action, I am refreshing the permissions using a isoformat date

Reply:
These have been updated.